enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+  - version: 4.4.1
+    image: quay.io/openshift-release-dev/ocp-release:4.4.1-x86_64
+  - version: 4.4.2
+    image: quay.io/openshift-release-dev/ocp-release:4.4.2-x86_64
+```
+
+Entries will be added to this list by the CLI tooling below, however an administrator could do this by hand if desired.
+
+If the CCO is enabled per normal best practice install, this field will be ignored, and should be absent.
+
+However if CCO is disabled, it will compare the ClusterVersion.Spec.AvailableUpdates to this list of ReconciledAvailableUpdates. If *any* available update is not present in the reconciled list, the CCO will report Upgradable=False.
+
+### Initial Installation
+
+```bash
+$ oc adm credentials reconcile --manifests=/path/to/installer/manifests
+```
+
+This command would parse the CredentialsRequests in the manifests directory, mint credentials in the cloud using the local users environment variables or credentials file, store the status on the CredentialsRequsts (which are yet to be created), write Secret manifests containing the cloud credentials, and disable the CCO.
+
+### Upgrade Reconciliation
+
+```bash
+$ oc adm credentials reconcile --available-updates
+```
+
+This command will iterate all currently available updates in ClusterVersion.AvailableUpdates. For each given release image we will extract each CredentialsRequest and use the local users environment variables/credentials file to compare and adjust permissions as necessary.
+
+This command reconciles permissions *in an additive fashion*. We do not want to reconcile any reduction in permissions as that would affect the running components in the cluster right now. The command would print a warning if this situation is encountered, and it can be resolved after upgrade with the next command.
+
+For net new CredentialsRequests in the update, we have a problem. We cannot assume to store a secret in the target namespace as it may be for a component that does not exist in the currently running OpenShift release. To work around this problem we propose copying net new CredentialsRequests into the cluster (all CredentialsRequests live in the openshift-cloud-credential-operator namespace today), and introducing the concept of temporary secret storage. This would allow us to mint the credentials when we have the root creds available, grab the secrets which we typically only get at creation time and cannot retrieve later, and transfer them to the correct namespace when it is available.
+
+Each available update we process successfully is added to the reconciledAvailableUpdates list.
+
+### On-Going Reconciliation
+
+```bash
+$ oc adm credentials reconcile
+```
+
+This command would re-reconcile against the CredentialsRequests currently in the cluster. This is more of an on-going configuration management to ensure something doesn't change unexpectedly on us due to external factors. It would however support reducing permissions as a result of an additive only reconcile pre-upgrade.
```

Comment:
+1 I was wondering how we get to the "minimum level of perms" after an upgrade is complete. This makes it clear how to drop permissions over time (assuming that will ever happen).